docs: add summary of goals (#3)
asyncapi/event-gateway: README.md

It supports all the protocols AsyncAPI supports through [bindings](https://github.com/asyncapi/bindings).

This **Event Gateway** is also compatible with the HTTP protocol, natively or through an external provider like [Krakend.io](http://krakend.io).

> AsyncAPI also offers a Marketplace where users can install middlewares for the **Event Gateway** made by the community or officially maintained by the AsyncAPI org.

Reviewer: is this correct and valid atm? what marketplace? or you just mean a list of official and community middleware? like we have with generator templates, that there are official ones and not

Marketplace kinda implies monetization, no?

Author: > Marketplace kinda implies monetization, no?

Yeah, you are right. I believe Marketplace implies monetization. 

> or you just mean a list of official and community middleware?

Yes, I do. WDYT about using the word `Catalog` instead?

Reviewer: `Catalog`? perfect 🚀

Author: I ended up removing the first mention to the Marketplace in favor of just the line we are talking about. Then simplified to `A catalog of middlewares made by the community is also available.`. It makes it simpler.

# Goals

## 1. Performance-first while ensuring message delivery

> The **Event Gateway** is a **stateless** solution that ensures messages are delivered as fast as possible using a minimal resource footprint.

Reviewer: `minimal resource footprint` ❤️

Author: Go Green 💚 !

Now I'm thinking if `minimal runtime footprint` would fit better.

Reviewer: tbh, I saw it more as a plus because of the cost, not going green 😄 `resource` is better I think

> The **Event Gateway** it's a **stateless** solution that ensures messages are delivered as fast as possible using a minimal resource footprint.

Reviewer 2: ```suggestion
The **Event Gateway** is a **stateless** solution that ensures messages are delivered as fast as possible using a minimal resource footprint.
```

Delivering messages as a top priority means no data loss should happen.

## 2. Transparent usage.

> No change in the user's code is needed. The service reads and responds pretending to be the final broker. 

Reviewer: `The service reads and responds pretending to be the final broker` you mean like a proxy, like a side car? right?

I'm kinda afraid of the work `pretending` 😄

Author: What about:

`The service acts as a proxy between the client and the final broker(s)`

Reviewer: yes, having the word `proxy` clears out a lot

Author: fixed!

> Messages infer the protocol based on the shape of the input network packet.

Reviewer 2: This is going to be interesting 😄

Author: Interesting and challenging, especially considering the first bullet.
However, remember those are just the goals based on a vision of how I see this service should be in the future (I would say years).
We can always modify them in case we see any of those are not of our interest anymore (For example, technically not worth the effort).

## 3. Fully configurable.
The service is entirely configurable and the user can specify the settings for all protocols as well. For example, consumers' and producers' settings.

## 4. API-first
The service provides an API for uploading AsyncAPI specs, allowing the user to update their message validation, among others, very quickly. 
It could even be an automated task whenever you update your specs.

## 5. Extensible

> The **Event Gateway** can extend its functionality via middlewares written by the community.
> A Marketplace is also available.

Reviewer 3: I am not a big fan of the new line 🤔 Maybe something like this?
```suggestion
The **Event Gateway** can extend its functionality via middlewares written by the community through a marketplace.
```

Author: I think newline makes it easier to read and understand. In fact, I'm just following https://sembr.org.
However, I have not strong opinion.
